### PR TITLE
cleanup(libsinsp): fix sinsp_ppm_sc bitmask checks

### PR DESCRIPTION
**Pull request by incertum**
Signed-off-by: Melissa Kilby <melissa.kilby.oss@gmail.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

> /area libpman

/area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fix bitmask checks in sinsp_ppm_sc `enforce_*_ppm_sc_set` methods (@FedeDP ).

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Andreagit97** [comment]
First of all, thank you for this! This PR fixes some bugs so I think we can merge it. One note that I would add is that this approach of using the `m_syscall_info_table` to retrieve the event category is not the right one in my opinion, I know that is the one that we use in all methods, but it is quite misleading. I've tried to explain all the reasons, in this PR https://github.com/falcosecurity/libs/pull/816. 

tl; dr; I think that right now it could be ok like that, but in the next future we need to find another way to recover this info from the event table as suggested here https://github.com/falcosecurity/libs/pull/816#discussion_r1056536959

WDYT? @FedeDP

File: userspace/libsinsp/sinsp_ppm_sc.cpp [hunk 1]
```diff
@@ -166,52 +166,73 @@ std::unordered_set<uint32_t> sinsp::enforce_simple_ppm_sc_set(std::unordered_set
...
 {
-	for(int i = 0; i < PPM_SC_MAX; i++)
+	const int bitmask = EC_SYSCALL - 1;
+	for(int ppm_sc_code = 0; ppm_sc_code < PPM_SC_MAX; ppm_sc_code++)
 	{
-		if(g_infotables.m_syscall_info_table[i].category == EC_IO_READ ||
-		   g_infotables.m_syscall_info_table[i].category == EC_IO_WRITE ||
-		   g_infotables.m_syscall_info_table[i].category == EC_IO_OTHER ||
-		   g_infotables.m_syscall_info_table[i].category == EC_FILE)
+		switch(g_infotables.m_syscall_info_table[ppm_sc_code].category & bitmask)
```
Comment (FedeDP) [inline userspace/libsinsp/sinsp_ppm_sc.cpp:172]: I think we could just use `& EC_IO_BASE` right? Like
```
for(int i = 0; i < PPM_SC_MAX; i++)
	{
		if(g_infotables.m_syscall_info_table[i].category & EC_IO_BASE)
```

**Review by FedeDP — COMMENTED** [review]
Changes LGTM! Nice catch!
We should really really really fix the categories madness though :/

File: userspace/libsinsp/sinsp_ppm_sc.cpp (hunk 1 above)
Comment (FedeDP) [reply userspace/libsinsp/sinsp_ppm_sc.cpp:172]: But i am ok with the changes, since we are not perf driven here, therefore it is ok to follow the same code logic of the other helpers.

Comment (incertum) [reply userspace/libsinsp/sinsp_ppm_sc.cpp:172]: First tried this type of approach and it wasn't working consistently, it tripped me off in the other PR. Then stumbled across https://github.com/falcosecurity/libs/blob/38cd13866be524e1c50058bc44d38598217dfd09/userspace/libsinsp/event.h#L253 and this approach always returns correct results.

**Comment by incertum** [comment]
> First of all, thank you for this! This PR fixes some bugs so I think we can merge it. One note that I would add is that this approach of using the `m_syscall_info_table` to retrieve the event category is not the right one in my opinion, I know that is the one that we use in all methods, but it is quite misleading. I've tried to explain all the reasons, in this PR #816.
> 
> tl; dr; I think that right now it could be ok like that, but in the next future we need to find another way to recover this info from the event table as suggested here [#816 (comment)](https://github.com/falcosecurity/libs/pull/816#discussion_r1056536959)


Added some comments in linked PR.

File: userspace/libsinsp/sinsp_ppm_sc.cpp (hunk 1 above)
Comment (FedeDP) [reply userspace/libsinsp/sinsp_ppm_sc.cpp:172]: Very weird about `EC_IO_BASE`; which issues have you seen?

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve
